bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Shell.java

```diff
@@ -1586,7 +1586,8 @@ public void setLocation(Point location) {
 
 @Override
 public void setLocation(int x, int y) {
-	Point location = display.translateLocationInPixelsInDisplayCoordinateSystem(x, y);
+	Point translatedPoint = display.translatePointIfInDisplayCoordinateGap(x, y);
```

Comment:
Isn't this something that should be hidden in the Display class, i.e., performed implicitly when calling `translateLocationInPixelsInDisplayCoordinateSystem`? Otherwise consumers of the display coordinate system need to know about this potential gap and handle it manually instead of having the display class guarantee the calculation of valid coordinates.